Library/Homebrew/test/caveats_spec.rb

```diff
@@ -26,4 +26,39 @@ def caveats
       expect(described_class.new(f)).not_to be_empty
     end
   end
+
+  describe "#caveats" do
+    describe "when f.plist is not null" do
+      it "when plist_startup is not null" do
+        allow(f).to receive(:plist).and_return(true)
+        allow(f).to receive(:plist_startup).and_return(true)
+        expect(subject.caveats).to include("To have launchd start #{f.full_name} now and restart at startup:\n  sudo brew services start #{f.full_name}\n")
+      end
+
+      it "when f.plist_startup is null" do
+        allow(f).to receive(:plist).and_return(true)
+        expect(subject.caveats).to include("To have launchd start #{f.full_name} now and restart at login:\n  brew services start #{f.full_name}\n")
+      end
+    end
+
+    describe "when f.keg_only is not null" do
+      it "" do
+        allow(f).to receive(:keg_only?).and_return(true)
+        expect(subject.caveats).to include("This formula is keg-only, which means it was not symlinked into #{HOMEBREW_PREFIX},\nbecause #{f.keg_only_reason.to_s.chomp}.")
+      end
+
+      it "f.bin.directory? is true" do
+        allow(f).to receive(:keg_only?).and_return(true)
+        bin = f.bin
+        allow(bin).to_receive(:directory?).and_return(true)
```

Comment:
Are you able to move this and the previous line onto one line, like you do in the `sbin` test?